compiler/rustc_hir_analysis/src/collect/resolve_bound_vars.rs

```diff
@@ -299,19 +266,10 @@ fn resolve_bound_vars(tcx: TyCtxt<'_>, local_def_id: hir::OwnerId) -> ResolveBou
         hir::OwnerNode::Synthetic => unreachable!(),
     }
 
-    let defs = named_variable_map.defs.into_sorted_stable_ord();
```

Comment:
The sorting that was done here...does that matter? I see `ResolveBoundVars` has SortedMaps, so just checking if that's enough? (Essentially I'm not sure why there was a need to sort this explicitly, I would have expected that you could build the SortedMap here and it would sort...)

Reply:
Yes, the sorting does matter. `UnordMap` has no ordering, it's a wrapper around a hashmap. To convert it into a `SortedMap` (which is a wrapper around `Vec`) we use `SortedMap::from_presorted_elements`, which requires a sorted `Vec` as input.

With the new code we build the `SortedMap` by inserting one item at a time, and it maintains its sortedness as we do that. This is likely to be slower, but this code isn't hot enough for that to matter.

Reply:
I guess, the question was not "does the sorting matter" but "does it matter that the sorting was done here in this fashion".


> With the new code we build the `SortedMap` by inserting one item at a time, and it maintains its sortedness as we do that. This is likely to be slower, but this code isn't hot enough for that to matter.

This address my question though.